models/discordactions.js

```diff
@@ -114,32 +116,56 @@ const updateDiscordImageForVerification = async (userDiscordId) => {
   }
 };
 
-const getNumberOfMemberForGroups = async (groups = []) => {
+/**
+ * Enriches group data with membership information for a given Discord ID.
+ *
+ * @param {string} discordId - The Discord ID of the user.
+ * @param {Array<object>} groups - Array of group objects to process.
+ * @returns {Promise<Array<object>>} - An array of group objects with enriched information.
+ */
+const enrichGroupDataWithMembershipInfo = async (discordId, groups = []) => {
   try {
     if (!groups.length) {
       return [];
     }
-    const roleIds = groups.map((group) => group.roleid);
 
+    const groupCreatorIds = groups.reduce((ids, group) => {
+      if (group.createdBy) {
+        ids.add(group.createdBy);
+      }
+      return ids;
+    }, new Set());
+
+    const groupCreatorsDetails = await retrieveUsers({ userIds: Array.from(groupCreatorIds) });
+
+    const roleIds = groups.map((group) => group.roleid);
     const snapshots = await memberRoleModel.where("roleid", "in", roleIds).get();
```

Comment:
Firesotore has a limitation of (10 or 30, I don't remember exactly),  how will this be handled here?

Reply:
https://firebase.google.com/docs/firestore/query-data/queries#limitations_2

Reply:
I was not aware of this. 
I will check this scenario and get back to you.

Reply:
fixed